File: C++_Programming/Practise_Programs/PP_033.cpp
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_033
//
//  Description       : Finds and displays the largest number
//                      present in an integer array.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 17/09/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

int main() 
{
    int numbers[5] = {12, 45, 7, 93, 28};
    
    int maxNumber = numbers[0];

    for (int i = 1; i < 5; ++i) 
    {
        if (numbers[i] > maxNumber) 
        {
            maxNumber = numbers[i]; 
        }
    }

    cout << "The largest number in the array is: " << maxNumber << endl;

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//      The largest number in the array is: 93
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_033.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_033
//
//  Description       : This program accepts a number from the user
//                      and calculates the sum of all digits present
//                      in the given number.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 10/07/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class DigitX
{
    //////////////////////////////////////////////////////////////////
    //
    //  Method Name      : SumDigits()
    //
    //  Description      : It is used to calculate the sum of all
    //                     digits present in the given number by
    //                     extracting each digit and adding it
    //                     to the sum.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 10/07/2026
    //
    //////////////////////////////////////////////////////////////////

    public int SumDigits(int iNo)
    {
        int iDigit = 0;
        int iSum = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10;
            iSum = iSum + iDigit;
            iNo = iNo / 10;
        }

        return iSum;
    }
}

class PP_033
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0;
        int iRet = 0;

        System.out.println("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = dobj.SumDigits(iValue);

        System.out.println("Summation of Digits: " + iRet);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter Number :
//      12345
//
//  Sample Output :
//      Summation of Digits: 15
//
//////////////////////////////////////////////////////////////////
